src/c/runtime/advice_chrome_tracing.cpp

```diff
@@ -0,0 +1,256 @@
+
+    pid = getpid ();
+    tid = my_gettid ();
+    cxa = abi::__cxa_demangle (fn, 0, 0, &status);
+    demangle = (cxa)? cxa : fn;
+    std::size_t pos = demangle.find_first_of ('(');
+    if (pos != std::string::npos)
+        demangle = demangle.substr (0, pos);
+
+    if (!(json = json_pack ("{s:s s:s s:I s:I s:f}",
+                                "name", demangle.c_str (),
+                                "cat", mn,
+                                "pid", static_cast<int64_t> (pid),
+                                "tid", static_cast<int64_t> (tid),
+                                "ts", ts))) {
+        free (cxa);
+        errno = ENOMEM;
+        return -1;
+    }
+    *o = json;
+    free (cxa);
+    return 0;
+}
+
+int advice_chrome_tracing_t::encode_event (json_t *o, const char *ph,
+                                           const char *scope, int64_t id)
+{
+    if (!o || !ph) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    json_t *ph_o;
+    if (!(ph_o = json_string (ph)))
+        goto json_memerror;
+    if (json_object_set_new (o, "ph", ph_o) < 0)
+        goto json_memerror;
+    if (scope) {
+        json_t *sc;
+        if (!(sc = json_string (scope)))
+            goto json_memerror;
+        if (json_object_set_new (o, "scope", sc) < 0)
+            goto json_memerror;
+    }
+    if (id >= 0) {
+        json_t *id_o;
+        if (!(id_o = json_integer (static_cast<json_int_t> (id))))
+            goto json_memerror;
+        if (json_object_set_new (o, "id", id_o) < 0)
+            goto json_memerror;
+    }
+    return 0;
+
+json_memerror:
+    errno = ENOMEM;
+    return -1;
+}
+
+
+/******************************************************************************
+ *                                                                            *
+ *             Public Methods of advice_chrome_tracing_t Class                *
+ *                                                                            *
+ ******************************************************************************/
+
+// Can throw std:::system_error or std::ofstream::failure exceptions
+advice_chrome_tracing_t::advice_chrome_tracing_t ()
+{
+    // TODO: add support for PERFLOW_OPTIONS
+    // TODO: especially PERFLOW_OPTIONS="log_file=my_name.log"
+    // TODO: support for TOML config
+    char hn[128];
+    if (gethostname (hn, 128) < 0)
+        throw std::system_error (errno,
+                                 std::system_category (),
+                                 "gethostname failed");
+    std::string fn = std::string ("perfflow.out.")
+                         + hn + std::string (".")
+                         + std::to_string (getpid ());
+    pid_t tid = my_gettid ();
+    m_before_counter_mutex = PTHREAD_MUTEX_INITIALIZER;
+    m_after_counter_mutex = PTHREAD_MUTEX_INITIALIZER;
+    m_ofs.open (fn, std::ofstream::out);
+    m_ofs << "[" << std::endl;
+}
+
+advice_chrome_tracing_t::~advice_chrome_tracing_t ()
+{
+}
```

Comment:
CTF doesn't require the closing bracket. But it is an excellent point to use the destructor to do some cleaning+optimization. I will add buffering (so that the trace will go memory until it hit a certain size) and as part of that I will flush the remaining buffer and close the output file in the destructor.